e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml

```diff
@@ -109,80 +123,174 @@
             ns: "{{ namespace }}"
             lkey: name
             lvalue: percona
+         when:  storage_engine  == 'jiva'
+
+       - name: 3b) Check if the cStor target has been created and running.
+         shell: source ~/.profile; kubectl get pods -n {{ operator_ns }} | grep {{ namespace }}
+         args:
+           executable: /bin/bash
+         register: cstor_pod
+         until: "'Running' in cstor_pod.stdout"
+         delay: 30
+         retries: 15
+         when: storage_engine  == 'cStor'
 
        - name: Wait for 120s to ensure liveness check starts
          wait_for:
            timeout: 120
 
-       - name: Get the name of the volume ctrl deployment
-         shell: >
-           source ~/.profile; kubectl get deployments -n {{ namespace }}
-           -l openebs.io/controller=jiva-controller --no-headers
-         args:
-           executable: /bin/bash
-         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
-         register: result_deploy
+       - block:
```

Comment:
Can we pass the controller/target deployment and labels to the subsequent tasks (resourceVersion identification/comparison, chaos etc.,) instead of repeating them twice for jiva and cstor ? (these can be set as facts based on storage_engine)